WalletWasabi.Fluent/ViewModels/Wallets/Buy/OrderViewModel.cs

```diff
@@ -102,19 +101,19 @@ public OrderViewModel(UiContext uiContext, IWalletModel wallet, Workflow workflo
 		_cts = new CancellationTokenSource();
 
 		// Handle Workflow Step Execution Errors and show UI message
-		Observable.FromEventPattern<Exception>(Workflow, nameof(Workflow.OnStepError))
-				  .DoAsync(async e =>
-				  {
-					  if (e.EventArgs is OperationCanceledException)
-					  {
-						  // Ignore
-					  }
-					  else
-					  {
-						  await ShowErrorAsync("Error while processing order.");
-					  }
-				  })
-				  .Subscribe();
+		Observable
+			.FromEventPattern<Exception>(Workflow, nameof(Workflow.OnStepError))
+			.DoAsync(async e =>
+			{
+				// Do not bother the user with an error dialog if they or not on BAB dialog.
+				if (uiContext.Navigate().DialogScreen.CurrentPage is not BuyViewModel)
```

Comment:
Alright let it be - but I think we will have problems with this in the future. Why is this running at all if Bab dialog is closed. It should be disposed at that time. 

![SquidwardSquidwardTentaclesGIF](https://github.com/zkSNACKs/WalletWasabi/assets/9844978/d22e88fe-0b8d-492f-ab94-a00ba2b1421a)

Reply:
The ViewModel must run always, because it keeps the Orders updated, which can sign to the user (with the unread badge) if there is a new message.

Reply:
ACK. Still, the solution seems a bit error-prone. So that is the best way to detect if the BaB dialog is open?

Reply:
Removed the check, I realized with the current logic an error dialog can only pop if the error happens in an interraction step. So that is expected.